miio/curtain_youpin.py

```diff
@@ -0,0 +1,290 @@
+    Open = 1
+    Close = 2
+    Auto = 3
+
+class Status(enum.Enum):
+    Stopped = 0
+    Opening = 1
+    Closing = 2
+
+class Polarity(enum.Enum):
+    Positive = 0
+    Reverse = 1
+
+class ManualEnabled(enum.Enum):
+    Disable = 0
+    Enable = 1
+
+class PosLimit(enum.Enum):
+    Unlimit = 0
+    Limit = 1
+
+class NightTipLight(enum.Enum):
+    Disable = 0
+    Enable = 1
+
+
+class CurtainStatus:
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """ Response from device
+            {'id': 1, 'result': [
+                {'did': 'motor_control', 'siid': 2, 'piid': 2, 'code': -4001},
+                {'did': 'current_position', 'siid': 2, 'piid': 3, 'code': 0, 'value': 0},
+                {'did': 'status', 'siid': 2, 'piid': 6, 'code': 0, 'value': 0},
+                {'did': 'target_position', 'siid': 2, 'piid': 7, 'code': 0, 'value': 0},
+                {'did': 'manual_enabled', 'siid': 4, 'piid': 1, 'code': 0, 'value': 1},
+                {'did': 'polarity', 'siid': 4, 'piid': 2, 'code': 0, 'value': 0},
+                {'did': 'position_limit', 'siid': 4, 'piid': 3, 'code': 0, 'value': 0},
+                {'did': 'night_tip_light', 'siid': 4, 'piid': 4, 'code': 0, 'value': 1},
+                {'did': 'run_time', 'siid': 4, 'piid': 5, 'code': 0, 'value': 0},
+                {'did': 'adjust_value', 'siid': 5, 'piid': 1, 'code': -4000}
+            ]}
+        """
+        self.data = data
+
+    @property
+    def status(self) -> Status:
+        """ Device status """
+        return Status(self.data["status"])
+
+    """ curtain_cfg """
+    @property
+    def manual_enabled(self) -> ManualEnabled:
+        """Manual control enable status
+        Values: Disable, Enable
+        """
+        return ManualEnabled(self.data["manual_enabled"])
+
+    @property
+    def polarity(self) -> Polarity:
+        """Polarity
+        Values: Positive, Reverse
+        """
+        return Polarity(self.data["polarity"])
+
+    @property
+    def position_limit(self) -> PosLimit:
+        """Position limit
+        Values: Unlimit, Limit
+        """
+        return PosLimit(self.data["position_limit"])
+
+    @property
+    def night_tip_light(self) -> NightTipLight:
+        """Night tip light status
+        Values: Enable, Disable
+        """
+        return NightTipLight(self.data["night_tip_light"])
+
+    @property
+    def run_time(self) -> int:
+        """Run time
+        Value range: [0, 255, 1]
+        """
+        return self.data["run_time"]
+
+    @property
+    def current_position(self) -> int:
+        """Current position
+        Value range: [0, 100, 1]
+        """
+        return self.data["current_position"]
+
+    @property
+    def target_position(self) -> int:
+        """Target position
+        Value range: [0, 100, 1]
+        """
+        return self.data["target_position"]
+
+    @property
+    def adjust_value(self) -> int:
+        """ Adjust value
+        Value range: [-100, 100, 1]
```

Comment:
```suggestion
```

Reply:
What does adjust value (and other properties, also) mean? The docstrings should be descriptive where possible :-)

Reply:
I think this is for the starting position limit. But right now I can't check it, because the curtain is in the process of installation.